### PR TITLE
perf: update dockerfile to better version

### PR DESCRIPTION
**Pull request by emonddr**
Update Dockerfile to a better version that is slimmer and has no vulnerabilities detected by IBM
Cloud

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

**Comment by emonddr** [comment]
IBM Cloud's Vulnerability Advisor detects security and configuration issues in docker images.
Needed to switch from node:10 to node:10.15-slim to avoid security and configuration issues, and to reduce the image size from 424 MB to 131 MB.

File: packages/cli/generators/app/templates/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:10
+FROM node:10.15-slim
```
Comment (raymondfeng) [inline packages/cli/generators/app/templates/Dockerfile]: Let's use `node:10-slim` to pick up latest 10.x version.

Comment (jannyHou) [reply packages/cli/generators/app/templates/Dockerfile]: Any reason change the variant to `slim`?

Comment (emonddr) [reply packages/cli/generators/app/templates/Dockerfile]: @jannyHou , the original setting (node:10) had security and configuration issues detected by IBM Cloud's vulnerability advisor (after pushing image to IBM Cloud registry). It was also 424 MB). The new 'slim' value is 131 MB.

Comment (raymondfeng) [reply packages/cli/generators/app/templates/Dockerfile]: To avoid confusions, we can probably add some comments to the generated `Dockerfile`.

Comment (emonddr) [reply packages/cli/generators/app/templates/Dockerfile]: @raymondfeng , ok I will try again as node:10-slim ...to see if there are any security or config issues....

Comment (emonddr) [reply packages/cli/generators/app/templates/Dockerfile]: @raymondfeng and I decided to use the dockerhub url instead: https://hub.docker.com/_/node

**Comment by emonddr** [comment]
@bajtos  your comment : 
"
Do we need to update any documentation files too? Isn't our Docker guide showing the example Dockerfile generated by lb4 app --docker?
"
The only place we showed the contents of a Dockerfile was in the tutorial 
https://loopback.io/doc/en/lb4/deploying_to_ibm_cloud_kubernetes.html . This tutorial was created before the `lb4 app --docker` option existed.  I have revised that tutorial in PR : https://github.com/strongloop/loopback-next/pull/2539, and we no longer show the contents of `Dockerfile`, but simply explain that it has been created for the user, and they should review the contents if it interests them.

**Comment by raymondfeng** [comment]
@emonddr When you quote a comment, please use [Markdown BlockQuote](https://guides.github.com/features/mastering-markdown/#syntax):

> @bajtos your comment :
Do we need to update any documentation files too? Isn't our Docker guide showing the example Dockerfile generated by lb4 app --docker?

**Comment by elv1s** [comment]
Why `-slim`? Does `-alpine` have vulnerabilities?

**Pull request merged**